src/Core/Domain/Carrier/Command/AddCarrierCommand.php

```diff
@@ -0,0 +1,203 @@
+     */
+    private $maxPackageDepth;
+
+    /**
+     * @var PackageWeightMeasure
+     */
+    private $maxPackageWeight;
+
+    /**
+     * @var int[]
+     */
+    private $associatedGroupIds;
+
+    /**
+     * @var int[]|null
+     */
+    private $associatedShopIds;
+
+    /**
+     * @param string[] $localizedName
+     * @param string[] $localizedDelay
+     * @param int $grade
+     * @param string $trackingUrl
+     * @param bool $includeShippingCost
+     * @param int $shippingMethod
+     * @param int $taxId
+     * @param bool $outOfRangeBehavior
+     * @param array $shippingRanges
+     * @param int $maxPackageWidth
+     * @param int $maxPackageHeight
+     * @param int $maxPackageDepth
+     * @param float $maxPackageWeight
+     * @param int[] $associatedGroupIds
+     * @param int[]|null $associatedShopIds
+     *
+     * @throws CarrierConstraintException
+     */
+    public function __construct(
+        array $localizedName,
+        array $localizedDelay,
+        $grade,
+        $trackingUrl,
+        $includeShippingCost,
+        $shippingMethod,
+        $taxId,
+        $outOfRangeBehavior,
+        array $shippingRanges,
+        $maxPackageWidth,
+        $maxPackageHeight,
+        $maxPackageDepth,
+        $maxPackageWeight,
+        array $associatedGroupIds,
+        array $associatedShopIds
+    ) {
+        $this->localizedName = $localizedName;
+        $this->localizedDelay = $localizedDelay;
+        $this->grade = new SpeedGrade($grade);
+        $this->trackingUrl = $trackingUrl;
+        $this->includeShippingCost = $includeShippingCost;
+        $this->shippingMethod = new ShippingMethod($shippingMethod);
+        $this->setShippingRanges($shippingRanges);
+        $this->taxId = $taxId;
+        $this->outOfRangeBehavior = $outOfRangeBehavior;
+        $this->maxPackageWidth = new PackageSizeMeasure($maxPackageWidth);
+        $this->maxPackageHeight = new PackageSizeMeasure($maxPackageHeight);
+        $this->maxPackageDepth = new PackageSizeMeasure($maxPackageDepth);
+        $this->maxPackageWeight = new PackageWeightMeasure($maxPackageWeight);
+        $this->associatedGroupIds = $associatedGroupIds;
+        $this->associatedShopIds = $associatedShopIds;
+    }
+
+    /**
+     * @param array $shippingRanges
+     */
+    private function setShippingRanges(array $shippingRanges)
+    {
+        foreach ($shippingRanges as $range) {
+            $resolvedRange = $this->resolveRangeParams($range);
+            $this->shippingRanges[] = new ShippingRange(
+                $resolvedRange['from'],
+                $resolvedRange['to'],
+                $resolvedRange['prices_by_zone_id']
+            );
+        }
+    }
+
+    /**
+     * Resolves array parameters to contain valid structure
+     *
+     * @param array $params
+     *
+     * @return array
+     */
+    private function resolveRangeParams(array $params)
+    {
+        $resolver = new OptionsResolver();
```

Comment:
interesting - should we not try to avoid symfony usage within the command at all cost? For such simple thing maybe simple validations here would be enough ?